### PR TITLE
Add code quality tools

### PR DESCRIPTION
**Pull request by ksuglobov**
This Pull Request includes the following changes to enhance and enforce code quality in the project:
- Added and configured code quality tools including flake8, black, pydocstyle, and isort.
- Integrated these tools with pipenv to manage dependencies and streamline usage.
- Introduced pre-commit hooks to automatically enforce code quality standards before commits.
- Applied necessary formatting fixes using pre-commit hooks.
- Updated the pre-commit configuration file to include the added tools.
- Excluded specific files from flake8 checks to prevent unnecessary warnings and errors.

These changes aim to ensure consistent code quality and formatting across the project by leveraging automated tools and pre-commit hooks.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[pydocstyle]
+convention = google
```
Comment (ilystsov) [inline setup.cfg:2]: I suggest using autodoc convention since we are using `Sphinx`

File: src/db/migration/env.py [hunk 2]
```diff
@@ -1,23 +1,20 @@
 import os
 from logging.config import fileConfig
 
-from dotenv import load_dotenv
-from sqlalchemy import engine_from_config
-from sqlalchemy import pool
-
 from alembic import context
-
-import src.db.models
```
Comment (ilystsov) [inline src/db/migration/env.py:10]: This import is used for alembic to find the db models. I suggest using `# noqa `on this line

File: src/vocabulary_builder.py [hunk 3]
```diff
@@ -45,35 +49,43 @@ def _(language: str):
...
-        name='index.html',
+        name="index.html",
         context=context,
     )
 
 
-@app.get('/new_word', response_class=JSONResponse)
-def get_new_word(db: Session = Depends(get_db)):
+@app.get("/new_word", response_class=JSONResponse)
+def get_new_word(db: Optional[Session] = None):
```
Comment (ilystsov) [inline src/vocabulary_builder.py:87]: It's better to ignore flake here and use `Depends`

**Review by ilystsov — COMMENTED** [review]
A very good PR, but there are some suggestions to consider!

File: setup.cfg (hunk 1 above)
Comment (ksuglobov) [reply setup.cfg]: Done

File: src/db/migration/env.py (hunk 2 above)
Comment (ksuglobov) [reply src/db/migration/env.py:10]: I added an exclusion to the flake8 configuration

File: src/vocabulary_builder.py (hunk 3 above)
Comment (ksuglobov) [reply src/vocabulary_builder.py]: Done

**Pull request merged**